robot/client/client.go

```diff
@@ -665,6 +668,7 @@ func (rc *RobotClient) resources(ctx context.Context) ([]resource.Name, []resour
 		if s, ok := status.FromError(err); !(ok && (s.Code() == codes.Unimplemented)) {
 			return nil, nil, err
 		}
+		rc.subtypesUnimplemented = true
```

Comment:
This should be inside of an `s.Code()==codes.Unimplemented` check. Otherwise an unrelated network error will set this flag and it will never be ran again.

Reply:
This already can only be reached when `s.Code() == Unimplemented`.

It will enter the `if` and return `err` when **either**:
- `!ok`
- `s.Code != Unimplemented`

Reply:
Can you add a comment above this line explaining why we're setting this flag?